tests/test_governance.py

```diff
@@ -45,3 +49,49 @@ def test_drop(self):
     def test_exec(self):
         # assert self.ds_pause.plot(self.plan).transact()
         assert self.ds_pause.exec(self.plan).transact()
+
+class TestDSChief:
+    def test_scenario(self, mcd: DssDeployment, our_address: Address, other_address: Address):
+        isinstance(mcd, DssDeployment)
+        isinstance(our_address, Address)
+
+        prevBalance = mcd.mkr.balance_of(our_address)
+        amount = Wad.from_number(1000)
+        mint_mkr(mcd.mkr, our_address, amount)
+        assert mcd.mkr.balance_of(our_address) == amount + prevBalance
+
+        # Lock MKR in DS-Chief
+        assert mcd.mkr.approve(mcd.ds_chief.address).transact(from_address=our_address)
+        assert mcd.ds_chief.lock(amount).transact(from_address=our_address)
+        assert mcd.mkr.balance_of(our_address) == prevBalance
+
+        # Vote for our address
+        assert mcd.ds_chief.vote_yays([our_address.address]).transact(from_address=our_address)
+        assert mcd.ds_chief.etch([other_address.address]).transact(from_address=our_address)
+
+        # Confirm that etch(our address) != etch(other address)
+        etches = mcd.ds_chief.past_etch(3)
+        assert etches[0].slate !=  etches[-1].slate
+
+        assert mcd.ds_chief.get_approvals(our_address.address) == amount
+
+        # Lift hat for our address
+        assert mcd.ds_chief.get_hat() != our_address
+        assert mcd.ds_chief.lift(our_address).transact(from_address=our_address)
+        assert mcd.ds_chief.get_hat() == our_address
+
+        # Now vote for other address
+        assert mcd.ds_chief.vote_etch(etches[-1]).transact(from_address=our_address)
+        assert mcd.ds_chief.lift(other_address).transact(from_address=our_address)
+        assert mcd.ds_chief.get_hat() == other_address
+
+        # TODO. Need to give DS-Chief approval to move/burn IOU tokens before
```

Comment:
In the future, I'd recommend splitting this out into a separate test, and then skipping that test.

Reply:
👍